src/ui/wxWidgets/properties.cpp

```diff
@@ -422,10 +429,68 @@ wxIcon CProperties::GetIconResource( const wxString& WXUNUSED(name) )
  * wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK
  */
 
-void CProperties::OnOkClick( wxCommandEvent& /* evt */ )
+void CProperties::OnOkClick( wxCommandEvent& WXUNUSED(evt) )
 {
 ////@begin wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK in CProperties.
   // Before editing this code, remove the block markers.
   EndModal(wxID_OK);
 ////@end wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK in CProperties.
 }
+
+/*!
+ * wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_CHANGE_NAME
+ */
+void CProperties::OnEditName( wxCommandEvent& WXUNUSED(evt) )
+{
+  wxTextEntryDialog textInputDialog(
+    this, _("Name:"), _("Please enter the new database name"), m_NewDbName.c_str(), wxOK|wxCANCEL
+  );
+
+  textInputDialog.SetSize(550, -1);
+
+  if (textInputDialog.ShowModal() == wxID_OK) {
+    m_DbName = textInputDialog.GetValue();
+
+    if (m_DbName.IsEmpty()) { /* Show 'N/A' on the UI in case of an empty string, */
+      m_DbName = _("N/A");    /* but use the empty string as new DB name          */
+
+      if (Validate() && TransferDataToWindow()) {
+        m_NewDbName = std2stringx(_T(""));
+      }
+    }
+    else {
+      if (Validate() && TransferDataToWindow()) {
+        m_NewDbName = std2stringx(m_DbName.wc_str());
```

Comment:
Could you please give me a short hint? Seems that I'm somehow blind for this.

Reply:
Something like this:
```c++
  if (textInputDialog.ShowModal() == wxID_OK) {
    auto newName = textInputDialog.GetValue();
    if (newName.IsEmpty()) { /* Show 'N/A' on the UI in case of an empty string, */
        newName = _T("");
        m_DbName = _("N/A");    /* but use the empty string as new DB name          */
    } else {
        m_DbName = newName;
    }
    if (Validate() && TransferDataToWindow()) {
        m_NewDbName = std2stringx(newName.wc_str());
    }
  }
```